src/common/com/intellij/plugins/haxe/model/type/HaxeTypeCompatible.java

```diff
@@ -141,6 +144,20 @@ else if(from.specifics.length == 0) {
       }
     }
 
+    // check interface/class type overrides
+    HaxeClass baseClass = to.clazz.getHaxeClass();
+    HaxeClass derivedClass = from.clazz.getHaxeClass();
+    if (baseClass != null && derivedClass != null) {
+      List<HaxeType> baseTypes = new ArrayList<HaxeType>();
+      baseTypes.addAll(derivedClass.getHaxeImplementsList());
+      baseTypes.addAll(derivedClass.getHaxeExtendsList());
+      for (HaxeType baseType : baseTypes) {
+        if(to.canAssign(HaxeTypeResolver.getTypeFromType(baseType))) {
```

Comment:
Miss your second example. So I will switch to collection of all available implementations.